extensions/elytron-security-ldap/deployment/src/test/java/io/quarkus/elytron/security/ldap/LdapSecurityRealmTest.java

```diff
@@ -0,0 +1,178 @@
+package io.quarkus.elytron.security.ldap;
+
+import static org.hamcrest.Matchers.equalTo;
+
+import java.net.InetAddress;
+
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import com.unboundid.ldap.listener.InMemoryDirectoryServer;
+import com.unboundid.ldap.listener.InMemoryDirectoryServerConfig;
+import com.unboundid.ldap.listener.InMemoryListenerConfig;
+import com.unboundid.ldif.LDIFReader;
+
+import io.quarkus.elytron.security.ldap.rest.ParametrizedPathsResource;
+import io.quarkus.elytron.security.ldap.rest.RolesEndpointClassLevel;
+import io.quarkus.elytron.security.ldap.rest.SingleRoleSecuredServlet;
+import io.quarkus.elytron.security.ldap.rest.SubjectExposingResource;
+import io.quarkus.elytron.security.ldap.rest.TestApplication;
+import io.restassured.RestAssured;
+
+/**
+ * Tests of BASIC authentication mechanism with the minimal config required
+ */
+
+public abstract class LdapSecurityRealmTest {
+
+    protected static Class[] testClasses = {
+            SingleRoleSecuredServlet.class, TestApplication.class, RolesEndpointClassLevel.class,
+            ParametrizedPathsResource.class, SubjectExposingResource.class
+    };
+
+    private static InMemoryDirectoryServer ldapServer;
+
+    @BeforeAll
+    public static void startLdapServer() throws Exception {
+        InMemoryListenerConfig listenerConfig = new InMemoryListenerConfig("listener", InetAddress.getByName("127.0.0.1"),
+                10389, null, null, null);
+
+        InMemoryDirectoryServerConfig inMemoryDirectoryServerConfig = new InMemoryDirectoryServerConfig("dc=quarkus,dc=io");
+        inMemoryDirectoryServerConfig.setListenerConfigs(listenerConfig);
+        inMemoryDirectoryServerConfig.addAdditionalBindCredentials("uid=admin,ou=system", "secret");
+        ldapServer = new InMemoryDirectoryServer(inMemoryDirectoryServerConfig);
+        ldapServer.importFromLDIF(true, new LDIFReader(ClassLoader.getSystemResourceAsStream("quarkus-io.ldif")));
+        ldapServer.startListening();
```

Comment:
I wonder if this should be implemented as a test resource instead? See what we did for H2 for instance (`H2DatabaseTestResource`). For now, the test resource could be in this artifact, no need to create a test artifact.

Reply:
I did this for the integration tests, but I did not test it with the unit tests.
I'm going to take another look and compact it. The ITs and unit tests are similar but I oriented on the elytron jdbc extension which also has overlapping tests.

Reply:
Outsourced this in an extra TestResource. Pushing soon.